public/app/core/components/RolePicker/RolePickerInput.tsx

```diff
@@ -60,7 +60,7 @@ export const RolePickerInput = ({
     <div className={styles.wrapper}>
       {showBasicRole && <ValueContainer>{basicRole}</ValueContainer>}
       {appliedRoles.map((role) => (
-        <ValueContainer key={role.uid}>{role.displayName}</ValueContainer>
+        <ValueContainer key={role.uid}>{role.displayName || role.name}</ValueContainer>
```

Comment:
this is not part of the move but in cases we do not have a displayname for the role this field would be empty